File: src/main/java/com/innercircle/survey/model/Question.java
package com.innercircle.survey.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Question extends BaseEntity {
    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    private Survey survey;

    private String question;

    private String description;

    @Enumerated(EnumType.STRING)
    private QuestionType questionType;

    private Boolean isRequired;

    @Setter
    @OneToMany(mappedBy = "question")
    private List<QuestionOption> questionOptions = new ArrayList<>();

    private Integer questionOrder;

    private Integer version;

    @Builder
    private Question(Survey survey, String question, String description, QuestionType questionType, Boolean isRequired, Integer questionOrder, Integer version) {
        this.survey = survey;
        this.question = question;
        this.description = description;
        this.questionType = questionType;
        this.isRequired = isRequired;
        this.questionOrder = questionOrder;
        this.version = version;
    }

    public void update(String question, String description, QuestionType questionType, Boolean isRequired, Integer version) {
        this.question = question;
        this.description = description;
        this.questionType = questionType;
        this.isRequired = isRequired;
        this.version = version;
    }
}

File: survey-domain/src/main/kotlin/com/innercircle/surveydomain/model/Question.kt
@file:Suppress("ktlint:standard:no-wildcard-imports")

package com.innercircle.surveydomain.model

import jakarta.persistence.*

@Entity
@Table(name = "questions")
data class Question(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    @ManyToOne
    @JoinColumn(name = "form_id", nullable = false)
    var form: Form,
    val formVersion: Int,
    val questionText: String,
    val questionType: String,
    @Column(name = "question_order", nullable = false)
    val questionOrder: Int,
    @Column(name = "is_required", nullable = false)
    val isRequired: Boolean,
    @Column(name = "additional_config")
    val additionalConfig: String?,
    @OneToMany(mappedBy = "question", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    val responses: List<Response> = emptyList(),
    @OneToMany(mappedBy = "question", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    var questionOptions: List<QuestionOption> = emptyList(),
)
